opal/datatype/opal_copy_functions_heterogeneous.c

```diff
@@ -89,13 +89,16 @@ struct bit80 {
 static inline void
 opal_dt_swap_long_double(void *to_p, const void *from_p, const size_t size, size_t count, uint32_t remoteArch)
 {
+
+#if defined(HAVE_IEEE754_H) && defined(__x86_64) || defined(__sparcv9)
```

Comment:
This is confusing - is there a way to make the precedence more obvious? I _think_ the "and" goes first and then the "or", but...

Reply:
Thanks - fixed.

Reply:
yeah this is wrong as && has a higher precendence than || so the new code is not equivalent to the old one. It should read  ```defined(HAVE_IEEE754_H) && (defined(__x86_64) || defined(__sparcv9))```.

Reply:
However this patch does not make the code more readable. We can remove the ```to``` (it is never used as a long double), and then we can move the declaration of i inside the for.

Reply:
thanks, updated.